### PR TITLE
Implement cgroups based memory pressure evaluator

### PR DESCRIPTION
**Pull request by spiiroin**

### CODE REVIEW

File: mce-common.c [hunk 1]
```diff
@@ -727,6 +731,59 @@ common_dbus_get_battery_level_cb(DBusMessage *const req)
...
+/** D-Bus callback for the get memory level method call
+ *
+ * @param msg The D-Bus message
+ *
+ * @return TRUE
+ */
+static gboolean
+common_dbus_get_memnotify_level_cb(DBusMessage *const req)
+{
+    mce_log(LL_DEVEL, "Received memory leve get request from %s",
```
Comment (sledges) [inline mce-common.c:765]: (old) typo `leve`

Comment (spiiroin) [reply mce-common.c]: fixed

File: modules/mempressure.c [hunk 2]
```diff
@@ -0,0 +1,754 @@
...
+
+    if( mempressure_level == prev )
+        goto EXIT;
+
+    mce_log(LL_WARN, "mempressure_level: %s -> %s",
+            memnotify_level_repr(prev),
+            memnotify_level_repr(mempressure_level));
+
+    datapipe_exec_full(&memnotify_level_pipe
+                       , GINT_TO_POINTER(mempressure_level));
```
Comment (mlehtima) [inline modules/mempressure.c:294]: A little inconsistent formatting, usually "," seems to be at the end of previous line.

File: modules/mempressure.c [hunk 3]
```diff
@@ -0,0 +1,754 @@
...
+{
+    (void) chn;
+    (void) aptr;
+
+    gboolean ret = G_SOURCE_REMOVE;
+
+    if( !mempressure_cgroup_event_id )
+        goto EXIT;
+
+    if( mempressure_cgroup_event_fd == -1 )
```
Comment (mlehtima) [inline modules/mempressure.c:353]: Would it make sense to compare these to one checking (mempressure_cgroup_event_id <= 0)?

File: modules/mempressure.c (hunk 2 above)
Comment (spiiroin) [reply modules/mempressure.c]: fixed

**Comment by spiiroin** [comment]
Review commits squashed.

**Review by mlehtima — APPROVED** [review]

**Pull request merged**